src/main/java/fr/insalyon/creatis/gasw/executor/slurm/SlurmMonitor.java

```diff
@@ -0,0 +1,159 @@
+import fr.insalyon.creatis.gasw.bean.Job;
+import fr.insalyon.creatis.gasw.dao.DAOException;
+import fr.insalyon.creatis.gasw.execution.GaswMonitor;
+import fr.insalyon.creatis.gasw.execution.GaswStatus;
+import fr.insalyon.creatis.gasw.executor.slurm.config.Constants;
+import fr.insalyon.creatis.gasw.executor.slurm.internals.SlurmJob;
+import fr.insalyon.creatis.gasw.executor.slurm.internals.SlurmManager;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j;
+
+@Log4j
+final public class SlurmMonitor extends GaswMonitor {
+
+    private static SlurmMonitor instance;
+
+    final private List<SlurmJob>      finishedJobs;
+    @Getter @Setter
+    private SlurmManager        manager;
+    private boolean 		    stop;
+
+    public synchronized static SlurmMonitor getInstance() {
+        if (instance == null) {
+            instance = new SlurmMonitor();
+            instance.start();
+        }
+        return instance;
+    }
+
+    private SlurmMonitor() {
+        super();
+        finishedJobs = new ArrayList<>();
+        stop = false;
+    }
+
+    private void statusChecker() {
+        final List<SlurmJob> jobs = manager.getUnfinishedJobs();
+
+        for (final SlurmJob j : jobs) {
+            final GaswStatus stus = j.getStatus();
+
+            System.err.println("job : " + j.getData().getJobID() + " : " + stus.toString());
+            if (stus != GaswStatus.RUNNING && stus != GaswStatus.QUEUED && stus != GaswStatus.UNDEFINED && stus != GaswStatus.NOT_SUBMITTED) {
+                j.setTerminated(true);
+                finishedJobs.add(j);
+            } else if (stus ==  GaswStatus.RUNNING) {
+                updateJob(j.getData().getJobID(), stus);
+            }
+        }
+    }
+
+    @Override
+    public void run() {
+        while (!stop) {
+            statusChecker();
+            try {
+                while (hasFinishedJobs()) {
+                    final SlurmJob sJob = pullFinishedJobID();
+                    final GaswStatus status = sJob.getStatus();
+                    final Job job = jobDAO.getJobByID(sJob.getData().getJobID());
+                    
+                    if (status == GaswStatus.ERROR || status == GaswStatus.COMPLETED) {
+                        job.setExitCode(sJob.getExitCode());
+                        job.setStatus(job.getExitCode() == 0 ? GaswStatus.COMPLETED : GaswStatus.ERROR);
+                    } else {
+                        job.setStatus(status);
+                    }
+                    System.err.println("job : " + sJob.getData().getJobID() + " final : " + job.getStatus());
+                    
+                    jobDAO.update(job);
+                    new SlurmOutputParser(sJob).start();
+                }
+
+                Thread.sleep(GaswConfiguration.getInstance().getDefaultSleeptime());
+
+            } catch (GaswException | DAOException | InterruptedException ex) {
+                log.error(ex);
```

Comment:
Add in log that it is ignored
Also InterruptedException should be handled differently and should not be ingored.